test/stablex/batch_exchange.js

```diff
@@ -1864,4 +1865,40 @@ contract("BatchExchange", async accounts => {
       assert.equal(await batchExchange.hasToken.call(erc20_1.address), true)
     })
   })
+  describe("Large Examples", () => {
+    it("ensures hard gas limit on largest possible ring trade ", async () => {
+      const batchExchange = await setupGenericStableX(25)
+      const sixMillion = 6000000
+      await makeDeposits(batchExchange, accounts, longRingTrade.deposits)
+      const batchId = (await batchExchange.getCurrentBatchId.call()).toNumber()
+      const orderIds = await placeOrders(batchExchange, accounts, longRingTrade.orders, batchId + 1)
+      await closeAuction(batchExchange)
+
+      const solution = solutionSubmissionParams(longRingTrade.solutions[0], accounts, orderIds)
+      const firstSubmissionTX = await batchExchange.submitSolution(
+        batchId,
+        solution.objectiveValue,
+        solution.owners,
+        solution.touchedorderIds,
+        solution.volumes,
+        solution.prices,
+        solution.tokenIdsForPrice,
+        { from: solver }
+      )
+      assert(firstSubmissionTX.receipt.gasUsed < sixMillion, "Solution submission exceeded 6 million gas")
```

Comment:
we could also limit the amount of gas on the tx to 6M but I like this better, can we even print the amount we used in case it fails?

Reply:
Done! Even before you mentioned!